src/Algebra/Construct/DirectProduct.agda

```diff
@@ -151,3 +161,73 @@ abelianGroup G H = record
     ; comm = λ x y → (G.comm , H.comm) <*> x <*> y
     }
   } where module G = AbelianGroup G; module H = AbelianGroup H
+
+semiringWithoutAnnihilatingZero : SemiringWithoutAnnihilatingZero a ℓ₁ →
+                                  SemiringWithoutAnnihilatingZero b ℓ₂ →
+                                  SemiringWithoutAnnihilatingZero (a ⊔ b) (ℓ₁ ⊔ ℓ₂)
+semiringWithoutAnnihilatingZero R S = record
+  { isSemiringWithoutAnnihilatingZero = record
+      { +-isCommutativeMonoid = CommutativeMonoid.isCommutativeMonoid
+                                  (commutativeMonoid R.+-commutativeMonoid
+                                                     S.+-commutativeMonoid)
+      ; *-isMonoid = Monoid.isMonoid (monoid R.*-monoid S.*-monoid)
+      ; distrib    = distribˡ , distribʳ
+      }
+  }
+  where
+  module R = SemiringWithoutAnnihilatingZero R
+  module S = SemiringWithoutAnnihilatingZero S
+  _≈_ = Pointwise R._≈_ S._≈_
+  _+_ = zip R._+_ S._+_
+  _*_ = zip R._*_ S._*_
+
+  distribˡ : _DistributesOverˡ_ _≈_ _*_ _+_
+  distribˡ (x , y) (z₁ , z₂) (u₁ , u₂) = R.distribˡ x z₁ u₁ , S.distribˡ y z₂ u₂
+
+  distribʳ : _DistributesOverʳ_ _≈_ _*_ _+_
+  distribʳ (x , y) (z₁ , z₂) (u₁ , u₂) = R.distribʳ x z₁ u₁ , S.distribʳ y z₂ u₂
+
+semiring : Semiring a ℓ₁ → Semiring b ℓ₂ → Semiring (a ⊔ b) (ℓ₁ ⊔ ℓ₂)
+semiring R S = record
+  { isSemiring = record
+      { isSemiringWithoutAnnihilatingZero =
+          SemiringWithoutAnnihilatingZero.isSemiringWithoutAnnihilatingZero U
+      ; zero = zeroˡ , zeroʳ
+      }
+  }
+  where
+  module R = Semiring R
+  module S = Semiring S
+  U = semiringWithoutAnnihilatingZero R.semiringWithoutAnnihilatingZero
+                                      S.semiringWithoutAnnihilatingZero
+  _≈_ = Pointwise R._≈_ S._≈_
+  zr  = R.0# , S.0#
+  _*_ = zip R._*_ S._*_
+
+  zeroˡ : LeftZero _≈_ zr _*_
+  zeroˡ (x , y) = R.zeroˡ x , S.zeroˡ y
+
+  zeroʳ : RightZero _≈_ zr _*_
+  zeroʳ (x , y) = R.zeroʳ x , S.zeroʳ y
+
+commutativeSemiring : CommutativeSemiring a ℓ₁ → CommutativeSemiring b ℓ₂ →
+                      CommutativeSemiring (a ⊔ b) (ℓ₁ ⊔ ℓ₂)
+commutativeSemiring R S = record
+  { isCommutativeSemiring = record
+      { isSemiring = record
```

Comment:
This can be written `isSemiring = Semiring.isSemiring (semiring R.semiring S.semiring)`